app/models/katello/repository.rb

```diff
@@ -123,6 +123,7 @@ class Repository < Katello::Model
     scope :archived, -> { where('environment_id is NULL') }
     scope :in_published_environments, -> { in_content_views(Katello::ContentView.non_default).where.not(:environment_id => nil) }
     scope :order_by_root, ->(attr) { joins(:root).order("#{Katello::RootRepository.table_name}.#{attr}") }
+    scope :with_content, ->(content) { joins(Katello::RepositoryTypeManager.find_content_type(content).model_class.repository_association_class.name.demodulize.underscore.pluralize.to_sym) }
```

Comment:
This looks good - but it needs a .distinct :+1:

Reply:
Added..